RELEASE_NOTES.md

```diff
@@ -1,5 +1,40 @@
 
+## [v4.0.0](https://github.com/mdn/browser-compat-data/releases/tag/v4.0.0)
+
+August 19, 2021
+
+Version 4.0.0 makes some potentially-breaking changes to improve the flexibility and quality of the data. Higlights include:
+
+- The schema now supports a `"preview"` version value for in-development browser releases, such as Safari Technology Preview, Chrome Canary, and Firefox Nightly.
+- Deno, the JavaScript and TypeScript runtime, has been added as a new engine.
+- All XPath (`xpath`) and XSLT (`xslt`) features have been removed because the data was incomplete and unmaintained.
+- The schema for browser release data now supports the optional `accepts_flags` boolean hint.
+
+### Known issues
+
+- In [v3.3.9](https://github.com/mdn/browser-compat-data/releases/tag/v3.3.9), the structure of features for `css.properties.place-content`, `place-items`, and `place-self` changed ([#11214](https://github.com/mdn/browser-compat-data/pull/11214)). This is an incomplete fix to the problem of undocumented namespaces appearing in places where you might otherwise expect compatability data. The restructuring of the data is a work in progress and may change again in a future release. See [#6175](https://github.com/mdn/browser-compat-data/issues/6175) for more information.
+
+### Notable changes
+
+- [Deno](https://deno.land/), the JavaScript and TypeScript runtime, has been added as a new engine. Deno data is included for JavaScript and some Web APIs, much like Node.js data. ([#10753](https://github.com/mdn/browser-compat-data/pull/10753))
+
+- The XPath (`xpath`) and XSLT (`xslt`) features have been removed. These features poorly covered their subjects and were unmaintained. Rather than purport to have such data, we removed it.
+
+  We're not ruling out the return of XPath and XSLT data in the future. But for now, in the absence of an intentional effort to create and maintain that data, we thought it best to remove the faulty data. ([#9830](https://github.com/mdn/browser-compat-data/pull/9830))
+
+- The schema now supports a special version value `"preview"` for in-development versions of browsers, such as Safari Technology Preview, Chrome Canary, and Firefox Nightly. Additionally, browser data now supports a `preview_name` property for supplying a corresponding name for the preview browser.
+
+  These additions to the schema should help represent features gated to development versions, which were previously misrepresented with notes, `false` values, or inaccurate version numbers. ([#10334](https://github.com/mdn/browser-compat-data/pull/10334))
+
+- The schema for browser release data now supports the optional `accepts_flags` boolean hint, which indicates whether a browser (or individual release) permits user-configurable preferences. Presently, this is a hint for data contributors or tools. The value of `accepts_flags` doesn't make any guarantees about the contents of feature data. ([#11286](https://github.com/mdn/browser-compat-data/pull/11286))
```

Comment:
>user-configurable preferences

User configurable or developer configurable? Google's flags are for developers, not users. I don't know if that's true on other browsers, but I'd like to know what you mean by this before I suggest something to clarify for all concerned.

(For my own education, "a hint for data contributors or tools" is vague to me. Do you have a concrete use case?)

Reply:
> Google's flags are for developers, not users.

This is a very fine distinction, so I've gone ahead and reworded it to avoid having to discuss it further. 😆

> For my own education, "a hint for data contributors or tools" is vague to me. Do you have a concrete use case?

I'd suggest reading the linked PR and https://github.com/mdn/browser-compat-data/issues/10616. My summary from that issue:

> Several of the browsers we track don't have meaningful support for flags, including Opera for Android, Samsung Internet, Firefox for Android (from version 79), and Android WebView. If we recorded this support explicitly in BCD, we could lint against incorrectly-introduced flag data and increase the accuracy browser data mirroring.

This paves the way for strict linting, but we can't actually enforce it just yet (there's a lot of bad flag data to remove). So, right now, it's a hint.